### PR TITLE
Multus metrics

### PR DESCRIPTION
**Pull request by aneeshkp**
Enables metrics to addmission contoller.
Metrics are available at :9091/metrtics

Desc:  Metric to count network attachment definitions.".
Name: network_attachment_definitions_count

Desc:Metric to get total pods using multus
Name: multus_pod_count

Desc:multus_enabled_instance_up (Either 0 or 1)
Name: Metric to identify clusters with multus enabled instances.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,6 +60,13 @@ EOF
 networkattachmentdefinition.k8s.cni.cncf.io/macvlan-conf created
 ```
 
+## Collecting metrics with Prometheus
+Network attachment definition admission controller comes with following metrics.
+  1. No of pod's with k8s.v1.cni.cncf.io/networks annotations
```
Comment (dougbtv) [inline README.md]: Nit: No need to abbreviate, you can use number or `No.` for an abbreviation

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
+// Copyright (c) 2018 Intel Corporation
```
Comment (dougbtv) [inline pkg/controller/controller.go]: You can update the date and say `2019 Network Plumbing Working Group`

File: pkg/localmetrics/localmetrics.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 RedHat
```
Comment (dougbtv) [inline pkg/localmetrics/localmetrics.go]: `2019 Network Plumbing Working Group`

File: pkg/controller/controller.go [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+const (
+	maxRetries       = 5
+	multusannotation = "k8s.v1.cni.cncf.io/networks"
```
Comment (dougbtv) [inline pkg/controller/controller.go]: Nit: Mind if we update this and other references to Multus to be generic? E.g. call it `annotationName` or something, as the admission controller itself is generic (doesn't require Multus)

File: README.md (hunk 1 above)
Comment (aneeshkp) [reply README.md]: Fixed

File: pkg/controller/controller.go (hunk 4 above)
Comment (aneeshkp) [reply pkg/controller/controller.go]: Fixed

File: pkg/controller/controller.go [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+}
+
+// Controller object
+type Controller struct {
+	clientset kubernetes.Interface
+	queue     workqueue.RateLimitingInterface
+	informer  cache.SharedIndexInformer
+}
+
+// Start prepares watchers and run their controllers, then waits for process termination signals
```
Comment (s1061123) [inline pkg/controller/controller.go]: Ditto (https://golang.org/doc/effective_go.html#commentary)

**Comment by aneeshkp** [comment]
@dougbtv @s1061123 , Added metrics for sriov and multus. Removed CR metrics, it was not accurate unless we introduced controller watching for net-attch-def resource.
 Following metrics are available for telemetry (check metrics doc  linked in readme ) 
The netowrks label will have unique network type names configured for a pod.
any= will give count for all types configured.  
list = will have unique n/w type name
sriov= is special , pick those pod that have sriov type configured
This will give you count for each network type
``` 
network_attachment_definition_instance_total{networks="any"}	5
network_attachment_definition_instance_total{networks="bridge,macvlan"}	2
network_attachment_definition_instance_total{networks="macvlan"}	3 
network_attachment_definition_instance_total{networks="sriov"}	3 
```

and This will indicate whether the cluster has enabled or not for network types
(here you will see one metric per type)

``` 
network_attachment_definition_enabled_instance_up{networks="any"}  1
network_attachment_definition_enabled_instance_up{networks="bridge"} 1
network_attachment_definition_enabled_instance_up{networks="macvlan"}  1
network_attachment_definition_enabled_instance_up{networks="sriov"}  1
``` 

I will be requesting network_attachment_definition_enabled_instance_up and network_attachment_definition_instance_total for telemetry. Hope thats good for this now.

**Pull request merged**